sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -220,3 +220,79 @@ def test_singular_values():
     rpca.fit(X_hat_rpca)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+
+
+def test_svd_whiten():
+    tsvd = TruncatedSVD(n_components=X.shape[1] - 1, random_state=42,
```

Comment:
@vene Thanks for the review. Yeah, it would have been nice to make `TruncatedSVD` not fail with `n_components=n_features`. Generally there is a lot of code overlap between `TruncatedSVD` and `PCA` maybe some of it could be factorized..